crates/noirc_frontend/src/hir/def_map/mod.rs

```diff
@@ -129,12 +129,23 @@ impl CrateDefMap {
     pub fn get_all_test_functions<'a>(
         &'a self,
         interner: &'a NodeInterner,
-    ) -> impl Iterator<Item = FuncId> + 'a {
+    ) -> impl Iterator<Item = TestFunction> + 'a {
         self.modules.iter().flat_map(|(_, module)| {
             module
                 .value_definitions()
                 .filter_map(|id| id.as_function())
-                .filter(|id| interner.function_meta(id).attributes == Some(Attribute::Test))
+                .filter(|id| {
+                    matches!(interner.function_meta(id).attributes, Some(Attribute::Test { .. }))
+                })
+                .map(|id| {
+                    if interner.function_meta(&id).attributes
+                        == Some(Attribute::Test { expect_failure: true })
```

Comment:
You can use a pattern match on this, and then pass `expect_failure` directly to the 2nd argument of `TestFunction::new`